appveyor: fix openssl version mismatch warning

ygj6 opened the pull request:
Many warnings appear when building and running with `Visual Studio 2019` in Appveyor:
> WARN C:\projects\libevent\test\regress_ssl.c:210: Version mismatch for openssl: compiled with 1000214f but running with 1000212f

Simply add the openssl binary to the "PATH" environment variable to fix it.

I wrote a simple demo to reproduce it: https://github.com/ygj6/verify
I see there are dozens of openssl libraries in the system of appveyor: https://ci.appveyor.com/project/ygj6/verify/builds/28290688
If you do not specify which openssl to use, the system cannot find the correct library.

azat commented:
So, can I merge this, or you are still doing something? (Since it is marked as draft)

ygj6 commented:
> So, can I merge this, or you are still doing something? (Since it is marked as draft)

I was waiting for CI checking, there are always some errors even still now.
Even though the "openssl version mismatch" warning disappeared, I'm not sure whether this PR should be merged since the appveyor has not really finished.

ygj6 commented:
I see you point. I tried to reset the repo to an earlier version, and it still fails on VS2019. It confused me. I‘m afraid I need more time to research.

azat commented:
>Interesting, I tried to RE-RUN INCOMPLETE on appveyor, but it constantly fails due to flacky tests for the VS2019 only, no idea how this can be related

Sigh, I looked too quickly, the test which fails is - `event_once_never`:

```
[00:05:24] main/event_once_never: [forking] [warn] evsig_init_: socketpair: Only one usage of each socket address (protocol/network address/port) is normally permitted.
[00:05:24] [msg] libevent using: win32
[00:05:24] [warn] event_base_new_with_config: Unable to make base notifiable.
[00:05:24]
[00:05:24]   [event_once_never FAILED]
```

ygj6 commented:
I'm still here.
I made a new [project](https://ci.appveyor.com/project/ygj6/libevent-92xnh) for libevent in my AppVeyor,  and it ran smoothly.  But it still failed in old project.

> main/event_once_never: [forking] [warn] evsig_init_: socketpair: Only one usage of each socket address (protocol/network address/port) is normally permitted.

I have seen this warning and it won't appear every time. Now that you mentioned it, I will look into it.

ygj6 commented:
When I changed `EVENT_TESTS_PARALLEL: 20` in the appveyor.yml file to `EVENT_TESTS_PARALLEL: 1`，that is, changed the command  from `ctest --output-on-failure -j20` to `ctest --output-on-failure -j1`, the pass rate of testing will be significantly improved. But the total test time is about 10 minutes longer than before. It works for both vs2017 and vs2019.

Maybe we can change to a single-threaded test to solve the problem of test failure.

ygj6 commented:
The test case `test-ratelim__group_lim` is highly likely to fail on Windows. If you change the parameter `check-grouplimit` from 1000 to 3000, the chances of success will increase greatly.

azat commented:
>When I changed EVENT_TESTS_PARALLEL: 20 in the appveyor.yml file to EVENT_TESTS_PARALLEL: 1，that is, changed the command from ctest --output-on-failure -j20 to ctest --output-on-failure -j1, the pass rate of testing will be significantly improved. But the total test time is about 10 minutes longer than before. It works for both vs2017 and vs2019.

Let's drop concurrency for vs2019 only for now?

azat commented:
>The test case test-ratelim__group_lim is highly likely to fail on Windows. 

Yeah, I saw these failures, but I cannot say that they are more frequent then regress failures.

> If you change the parameter check-grouplimit from 1000 to 3000, the chances of success will increase greatly.

As I didn't paying attention how this will affect the outcome, let me take a look

azat commented:
> > If you change the parameter check-grouplimit from 1000 to 3000, the chances of success will increase greatly.

> As I didn't paying attention how this will affect the outcome, let me take a look

Hm, even 1% of my laptop is enough to handle this particular bit of the test, will make a closer look

ygj6 commented:
> Let's drop concurrency for vs2019 only for now?

It also helps vs2017. I suggest dropping it for both vs2017 and vs2019.

azat commented:
>I suggest dropping it for both vs2017 and vs2019.

But this will increase running time a lot, but okay let's try this way
